partner_preview_scripts/ValidationLab.py

```diff
@@ -3,432 +3,472 @@
+                "CO": {"name": "PAN-CO", "description": "Pan-colorectal carcinoma"}
+            }
+        },
+        "MS_status": {
+            "direct_mapping": {
+                "MSI": {"name": "MSI", "description": "Microsatellite unstable"},
+                "MSS": {"name": "MSS", "description": "Microsatellite stable"},
+            }
+        },
+        "PAM50_status": {
+            "direct_mapping": {
+                "Luminal B": {
+                    "name": "Luminal B",
+                    "description": "PAM50 status: luminal B",
+                },
+                "Luminal A": {
+                    "name": "Luminal B",
+                    "description": "PAM50 status: luminal A",
+                },
+                "Basal": {"name": "Basal", "description": "PAM50 status: basal"},
+            }
+        },
+        "Hormone_status": {
+            "direct_mapping": {
+                "Hormone dependent": {
+                    "name": "Hormone dependent",
+                    "description": "Hormone dependency: hormone dependent",
+                },
+                "Hormone refractory (TNBC)": {
+                    "name": "Hormone refractory (TNBC)",
+                    "description": "Hormone dependency:  hormone refractory (triple-negative breast cancer)",
+                },
+            }
+        },
+        "CRIS_subtype": {
+            "direct_mapping": {
+                "A": {
+                    "name": "CRIS-A",
+                    "description": "mucinous, glycolytic, enriched for microsatellite instability or KRAS mutations.",
+                },
+                "B": {
+                    "name": "CRIS-B",
+                    "description": "TGF-β pathway activity, epithelial-mesenchymal transition, poor prognosis.",
+                },
+                "C": {
+                    "name": "CRIS-C",
+                    "description": "elevated EGFR signalling, sensitivity to EGFR inhibitors.",
+                },
+                "D": {
+                    "name": "CRIS-D",
+                    "description": "WNT activation, IGF2 gene overexpression and amplification.",
+                },
+                "E": {
+                    "name": "CRIS-E",
+                    "description": "Paneth cell-like phenotype, TP53 mutations.",
+                },
+                "?": {"name": "CRIS-?", "description": "CRIS subtype undetermined."},
+            }
+        },
+        "KRAS_status": {
+            "description": "KRAS mutation status: ",
+            "name": "KRAS-",
+        },
+        "TP53_status": {
+            "description": "TP53 mutation status: ",
+            "name": "TP53-",
+        },
+        "APC_status": {
+            "description": "APC mutation status: ",
+            "name": "APC-",
+        },
+        "BRAF_status": {"description": "BRAF mutation status: ", "name": "BRAF-"},
+    }
+
+    @staticmethod
+    def _get_biomarker_wrapper(biomarker_map: dict) -> Callable:
```

Comment:
I did this I think because I couldn't pass self into udf. But I have to double check.

Reply:
This thread says UDFs need to be static methods, because you only want to serialise a single function, not the function of an instance.
It could be as easy as defining the UDF as an static function like so.
I'd suggest trying, but of course it is not a blocker
```
@staticmethod
@f.udf(
            t.StructType(
                [
                    t.StructField("name", t.StringType(), False),
                    t.StructField("description", t.StringType(), False),
                ]
            )
        )
        def _get_biomarker_wrapper(column_name: Column, biomarker: Column) -> dict | None:
        ...
```

Reply:
Yes, exactly, but in that case you need to bake the mapping dictionary into the udf. I wanted to avoid that.